vault/token_store.go

```diff
@@ -596,6 +729,15 @@ func (ts *TokenStore) handleCreateCommon(
 		CreationTime: time.Now().Unix(),
 	}
 
+	if role != nil {
+		te.Role = role.Name
+
+		te.Path = fmt.Sprintf("%s/%s", te.Path, role.Name)
+		if role.Prefix != "" {
+			te.Path = fmt.Sprintf("%s/%s", te.Path, role.Prefix)
```

Comment:
The thought was that it's an extra prefix added on the path before the lease ID. But I can see why it's confusing. Maybe I should rename it to PathSuffix?